python-avd/pyavd/_eos_designs/structured_config/underlay/utils.py

```diff
@@ -250,60 +250,63 @@ def _get_l2_as_subint(
         link: dict,
         svi: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem.SvisItem,
         vrf: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem,
-    ) -> dict:
+    ) -> EosCliConfigGen.EthernetInterfacesItem:
         """
         Return structured config for one subinterface representing the given SVI.
 
         Only supports static IPs or VRRP.
         """
         is_native = svi.id == link.get("native_vlan")
         interface_name = link["interface"] if is_native else f"{link['interface']}.{svi.id}"
-        subinterface = {
-            "name": interface_name,
-            "peer": link["peer"],
-            "peer_interface": f"{link['peer_interface']} VLAN {svi.id}",
-            "peer_type": link["peer_type"],
-            "description": default(svi.description, svi.name),
-            "shutdown": not default(svi.enabled, False),  # noqa: FBT003
-            "switchport": {"enabled": False if is_native else None},
-            "encapsulation_dot1q": {"vlan": None if is_native else svi.id},
-            "vrf": vrf.name if vrf.name != "default" else None,
-            "ip_address": svi.ip_address,
-            "ipv6_address": svi.ipv6_address,
-            "ipv6_enable": svi.ipv6_enable,
-            "mtu": svi.mtu if self.shared_utils.platform_settings.feature_support.per_interface_mtu else None,
-            "eos_cli": svi.raw_eos_cli,
-            "flow_tracker": link.get("flow_tracker"),
-        }
+        subinterface = EosCliConfigGen.EthernetInterfacesItem(
+            name=interface_name,
+            peer=link["peer"],
+            peer_interface=f"{link['peer_interface']} VLAN {svi.id}",
+            peer_type=link["peer_type"],
+            description=default(svi.description, svi.name),
+            shutdown=not default(svi.enabled, False),  # noqa: FBT003
+            switchport=EosCliConfigGen.EthernetInterfacesItem.Switchport(enabled=False) if is_native else Undefined,
+            encapsulation_dot1q=EosCliConfigGen.EthernetInterfacesItem.EncapsulationDot1q(vlan=svi.id) if not is_native else Undefined,
+            vrf=vrf.name if vrf.name != "default" else None,
+            ip_address=svi.ip_address,
+            ipv6_address=svi.ipv6_address,
+            ipv6_enable=svi.ipv6_enable,
+            mtu=svi.mtu if self.shared_utils.platform_settings.feature_support.per_interface_mtu else None,
```

Comment:
should we use Undefined instead of None where possible? (same for vrf)

```suggestion
            mtu=svi.mtu if self.shared_utils.platform_settings.feature_support.per_interface_mtu else Undefined,
```

Reply:
not needed for values